### PR TITLE
Wildcard V2: <Popover /> style migration

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
Manual Migrations to <Popover /> wildcard component, in highlighted files and components with usage of `dropdown-menu`

## Refs
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-27695)
[SourceGraph Link](https://github.com/sourcegraph/sourcegraph/issues/27695)
[Review Changes](https://www.chromatic.com/pullrequest?appId=5f0f381c0e50750022dc6bf7&number=30231&view=changes)
[Loom video](https://www.loom.com/share/9cf7574f320d43f1b8794167dbdd2a17)
[Parent Issue](https://github.com/sourcegraph/sourcegraph/issues/29200)

## Test Plan
 - Run web app using preview app.
 - Confirm there are no UI variations while comparing all use cases of Popover with prod environment

## Preview App
[Frontend](https://client-sourcegraph-frontend-pr-239.onrender.com/)
[Storybook](https://client-sourcegraph-storybook-pr-239.onrender.com/)

### Styles migration:
1. Migrate usage of `dropdown-menu` class to Wildcard Popover
1. Remove `dropdown-menu` Bootstrap dependency - reimplement styles in Popover CSS module

Note: `dropdown-menu` usage is also required for the `Menu` components. We should wait until this migration (https://github.com/sourcegraph/sourcegraph/issues/27701) is finished before migrating away from global styles.

## App preview:

- [Web](https://sg-web-contractors-sg-27695-1.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-mlyraxcmss.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, could you take a look at this PR? We still want to get rid of the Bootstrap import.

**Comment by valerybugakov** [comment]
Let's link to [a new parent issue](https://github.com/sourcegraph/sourcegraph/issues/29200) since [the Popover migration ticket](https://github.com/sourcegraph/sourcegraph/issues/27695#issuecomment-1131289656) is closed and the migration is completed.

File: client/web/src/nav/UserNavItem.tsx [hunk 1]
```diff
@@ -229,39 +237,51 @@ export const UserNavItem: React.FunctionComponent<React.PropsWithChildren<UserNa
...
-                            <MenuLink as={AnchorLink} to="https://about.sourcegraph.com" target="_blank" rel="noopener">
-                                About Sourcegraph <Icon role="img" as={OpenInNewIcon} aria-hidden={true} />
+                            <MenuLink
+                                className={styles.dropdownItem}
+                                as={AnchorLink}
+                                to="https://about.sourcegraph.com"
+                                target="_blank"
+                                rel="noopener"
+                            >
+                                About Sourcegraph <Icon as={OpenInNewIcon} />
```
Comment (valerybugakov) [inline client/web/src/nav/UserNavItem.tsx:279]: Why did we remove accessibility attributes from the `Icon` here?

File: client/web/src/nav/UserNavItem.tsx [hunk 2]
```diff
@@ -229,39 +237,51 @@ export const UserNavItem: React.FunctionComponent<React.PropsWithChildren<UserNa
...
                         {props.authenticatedUser.siteAdmin && (
-                            <MenuLink as={Link} to="/site-admin">
+                            <MenuLink className={styles.dropdownItem} as={Link} to="/site-admin">
                                 Site admin
                             </MenuLink>
                         )}
-                        <MenuLink as={Link} to="/help" target="_blank" rel="noopener">
-                            Help <Icon role="img" as={OpenInNewIcon} aria-hidden={true} />
+                        <MenuLink className={styles.dropdownItem} as={Link} to="/help" target="_blank" rel="noopener">
+                            Help <Icon as={OpenInNewIcon} />
```
Comment (valerybugakov) [inline client/web/src/nav/UserNavItem.tsx:261]: Why did we remove accessibility attributes from the `Icon` here?

**Review by valerybugakov — CHANGES_REQUESTED** [review]

File: client/web/src/nav/UserNavItem.tsx (hunk 1 above)
Comment (gitstart-sourcegraph) [reply client/web/src/nav/UserNavItem.tsx]: Resulted from wrong conflict resolution. Reverted...

File: client/web/src/nav/UserNavItem.tsx (hunk 2 above)
Comment (gitstart-sourcegraph) [reply client/web/src/nav/UserNavItem.tsx]: same as above

File: client/wildcard/src/components/Menu/MenuText.tsx [hunk 3]
```diff
@@ -4,14 +4,16 @@ import classNames from 'classnames'
...
+
 export interface MenuTextProps {}
 
 /**
  * A simple styled wrapper component that can be used
  * in and/or outside the Menu context.
  */
 export const MenuText = React.forwardRef(({ children, as: Component = 'div', className, ...props }, reference) => (
-    <Component role="menuitem" ref={reference} {...props} className={classNames('dropdown-item', className)}>
+    <Component ref={reference} {...props} className={classNames(styles.dropdownItem, className)}>
```
Comment (valerybugakov) [inline client/wildcard/src/components/Menu/MenuText.tsx:16]: Should we keep the `role` attribute here?

**Review by valerybugakov — APPROVED** [review]
Let's fix one more accessibility regression and merge this PR. 
Please keep an eye on unexpected changes like this.

**Pull request merged**